validation/hess-dl3-dr1/run.py

```diff
@@ -0,0 +1,213 @@
+
+    # Flux points
+    e_edges = (
+        np.logspace(np.log10(EMIN.value), np.log10(EMAX.value), N_FLUX_POINTS + 1)
+        * u.TeV
+    )
+    fpe = FluxPointsEstimator(datasets=datasets, e_edges=e_edges)
+    flux_points = fpe.run()
+    keys = ["e_ref", "e_min", "e_max", "dnde", "dnde_errp", "dnde_errn"]
+    ascii.write(
+        flux_points.table_formatted[keys], str(PATH_RESULTS / "flux-points-1d.dat")
+    )
+
+# Run 3D analysis
+for target in targets:
+    TAG = target["tag"]
+
+    PATH_CONFIG = Path(TAG + "/config/")
+    PATH_RESULTS = Path(TAG + "/results/")
+    EMIN, EMAX = [0.7, 60] * u.TeV  # Used for flux points estimation
+    N_FLUX_POINTS = 15
+
+    filename = str(PATH_CONFIG / "config_3d.yaml")
+    config = yaml.load(open(filename))
+    config = AnalysisConfig(config)
+
+    #  Observation selection
+    analysis = Analysis(config)
+    analysis.get_observations()
+
+    # Data reduction
+    analysis.get_datasets()
+
+    # Set runwise energy threshold. See reference paper, section 5.1.1.
+    for dataset in analysis.datasets:
+        # E_thr_bias (= 10% edisp)
+        geom = dataset.counts.geom
+        e_reco = geom.get_axis_by_name("energy").edges
+        E_thr_bias = dataset.edisp.get_bias_energy(0.1)
+
+        # E_thr_bkg (= background peak energy)
+        background_model = dataset.background_model
+        bkg_spectrum = background_model.map.get_spectrum()
+        peak = bkg_spectrum.data.max()
+        idx = list(bkg_spectrum.data).index(peak)
+        E_thr_bkg = bkg_spectrum.energy.center[idx]
+
+        esafe = max(E_thr_bias, E_thr_bkg)
+        dataset.mask_fit = geom.energy_mask(emin=esafe)
+
+    # Model fitting
+    model_config = str(PATH_CONFIG / "config_model.yaml")
+    analysis.set_model(filename=model_config)
+
+    for dataset in analysis.datasets:
+        dataset.background_model.norm.frozen = False
+        dataset.background_model.tilt.frozen = False
+
+    result = analysis.run_fit()
+    assert analysis.fit_result.success == True
+
+    # Store fit results
+    fit_results_dict = {}
+    parameters = analysis.model.parameters
+    parameters.covariance = analysis.fit_result.parameters.covariance[0:5, 0:5]
+    for parameter in parameters:
+        value = parameter.value
+        error = parameters.error(parameter)
+        unit = parameter.unit
+        name = parameter.name
+        string = "{0:.2e} +- {1:.2e} {2}".format(value, error, unit)
+        fit_results_dict.update({name: string})
+    with open(str(PATH_RESULTS / "results-summary-fit-3d.yaml"), "w") as f:
+        yaml.dump(fit_results_dict, f)
+
+    # Flux points
+    # TODO: This is a workaround to reoptimize the bkg in each energy bin. Add this option to the Analysis class
+    datasets = analysis.datasets.copy()
+    for dataset in datasets:
+        for par in dataset.parameters:
+            if par is not dataset.background_model.norm:
+                par.frozen = True
+
+    e_edges = (
```

Comment:
Suggest we always use `MapAxis` to create energy grids instead of `np.logspace`:
https://github.com/gammapy/gammapy-benchmarks/blob/a127199a726dee9040663744390fa1a1998c9fad/benchmarks/analysis_3d.py#L29
We can try to improve the API there to make it a bit more convenient, but for now it's already a bit nicer than having to use `np.logspace(np.log10(EMIN.value), ...`